File: model-client/src/main/java/org/modelix/model/lazy/TreeId.java
package org.modelix.model.lazy;

import java.util.UUID;
import org.jetbrains.annotations.NotNull;

public class TreeId {
  public static TreeId random() {
    return new TreeId(UUID.randomUUID().toString());
  }

  private final String id;

  public TreeId(@NotNull String id) {
    this.id = id;
  }

  public String getId() {
    return this.id;
  }

  @Deprecated
  public String getMasterBranchKey() {
    return getBranchKey(null);
  }

  public String getBranchKey(String branchName) {
    if ((branchName == null || branchName.length() == 0)) {
      branchName = "master";
    }
    return "branch_" + id + "_" + branchName;
  }

  @Override
  public boolean equals(Object o) {
    if (this == o) {
      return true;
    }
    if (o == null || this.getClass() != o.getClass()) {
      return false;
    }

    TreeId that = (TreeId) o;
    if ((id != null ? !(((Object) id).equals(that.id)) : that.id != null)) {
      return false;
    }

    return true;
  }

  @Override
  public int hashCode() {
    int result = 0;
    result = 31 * result + ((id != null ? String.valueOf(id).hashCode() : 0));
    return result;
  }

  @Override
  public String toString() {
    return id;
  }
}

File: model-client/src/main/kotlin/org/modelix/model/lazy/TreeId.kt
package org.modelix.model.lazy

import java.util.*

class TreeId(val id: String) {

    @get:Deprecated("")
    val masterBranchKey: String
        get() = getBranchKey(null)

    fun getBranchKey(branchName: String?): String {
        var branchName = branchName
        if (branchName == null || branchName.length == 0) {
            branchName = "master"
        }
        return "branch_" + id + "_" + branchName
    }

    override fun equals(o: Any?): Boolean {
        if (this === o) {
            return true
        }
        if (o == null || this.javaClass != o.javaClass) {
            return false
        }
        val that = o as TreeId
        return if (if (id != null) id as Any != that.id else that.id != null) {
            false
        } else true
    }

    override fun hashCode(): Int {
        var result = 0
        result = 31 * result + (id?.hashCode() ?: 0)
        return result
    }

    override fun toString(): String {
        return id
    }

    companion object {
        @JvmStatic
        fun random(): TreeId {
            return TreeId(UUID.randomUUID().toString())
        }
    }

}